torch/csrc/jit/passes/to_batch.cpp

```diff
@@ -0,0 +1,78 @@
+#include "torch/csrc/jit/passes/to_batch.h"
+#include "torch/csrc/jit/script/compiler.h"
+
+namespace torch { namespace jit {
+
+// map from batchTensor to {data, mask, dims}
+static std::unordered_map<Value*, std::vector<Value*>> batch_map;
+
+static void ToBatch(Block* block, Block* res_block) {
+  for (auto it = block->nodes().begin(); it != block->nodes().end(); it++) {
+    auto n = *it;
+    // replace tensor operator to BatchTensor operator
+    if(n->kind().is_aten()){
+      auto batch_graph = batch_operator_table.at(n->kind().toUnqualString());
+      WithInsertPoint guard(res_block);
+      std::vector<Value*> new_inputs;
+      for(Value *input : n->inputs()){
+        if(batch_map.find(input) != batch_map.end()){
+          auto new_input = batch_map.at(input);
+          new_inputs.insert(new_inputs.end(), new_input.begin(), new_input.end());
+        }
+        else{
+          new_inputs.push_back(input);
+        }
+      }
+      auto outputs = script::inlineCallTo(*res_block->owningGraph(), *batch_graph, new_inputs);
+      // Assume all outputs from inlined operator implementation are in the triple form.
+      for(size_t i = 0; i < n->outputs().size(); i++){
+        auto output = n->outputs()[i];
+        batch_map[output] = std::vector<Value*>(outputs.begin() + i * 3, outputs.begin() + i * 3 + 3);
+      }
+    }
+    // control flow: not supported yet, will be added further
```

Comment:
Anything not supported yet should throw an exception.  (e.g. "NYI: ..."). It will be hard to debug if the code fails silently.